site/app/controllers/banner/BannerController.php

```diff
@@ -0,0 +1,201 @@
+
+declare(strict_types=1);
+
+namespace app\controllers\banner;
+
+use app\controllers\AbstractController;
+use app\controllers\GlobalController;
+use app\libraries\response\JsonResponse;
+use app\libraries\response\WebResponse;
+use app\views\banner\BannerView;
+use Symfony\Component\Routing\Annotation\Route;
+use app\entities\banner\BannerImage;
+use app\libraries\DateUtils;
+use app\libraries\FileUtils;
+use app\libraries\routers\AccessControl;
+
+/**
+ * @AccessControl(level="FACULTY")
+ */
+class BannerController extends AbstractController {
+    /**
+     *
+     * @Route("/banner")
+     *
+     * @return WebResponse
+     * @see GlobalController::prep_user_sidebar
+     * @see BannerView::showEventBanners
+     */
+    public function viewCommunityEvents(): WebResponse {
+        $communityEventBanners = $this->core->getSubmittyEntityManager()->getRepository(BannerImage::class)->findAll();
+        return new WebResponse(BannerView::class, 'showEventBanners', $communityEventBanners);
+    }
+
+
+
+    /**
+     * @Route("/banner/upload", methods={"POST"})
+     */
+    public function ajaxUploadEventFiles(): JsonResponse {
+        $upload_path = FileUtils::joinPaths($this->core->getConfig()->getSubmittyPath(), "community_events");
+
+        if (isset($_POST['release_time'])) {
+            $release_date = DateUtils::parseDateTime($_POST['release_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+
+
+        if (isset($_POST['close_time'])) {
+            $close_date = DateUtils::parseDateTime($_POST['close_time'], $this->core->getDateTimeNow()->getTimezone());
+        }
+        else {
+            return JsonResponse::getErrorResponse("No release date.");
+        }
+        if (!isset($_FILES["files1"]) || $_FILES["files1"] === []) {
+            return JsonResponse::getErrorResponse("No files were submitted.");
+        }
+
+        $uploaded_files = $_FILES["files1"];
+        $count_item = count($uploaded_files["name"]);
+        $extra_name = $_POST['extra_name'];
+        $link_name = $_POST['link_name'];
+
+        if ($extra_name === "" && $count_item !== 1) {
+            return JsonResponse::getErrorResponse("You can only have one banner submitted.");
+        }
+        elseif ($count_item > 2) {
+            return JsonResponse::getErrorResponse("Can't have more than two banners submitted.");
+        }
+
+
+        $specificPath = $close_date->format("Y");
+        $actual_banner_name = "";
+
+        foreach ($uploaded_files['name'] as $uploaded_file_name) {
+            if ($uploaded_file_name !== $extra_name) {
+                $actual_banner_name = $uploaded_file_name;
+                break; // Exit the loop once a valid name is found
+            }
+        }
+
+        $currentDate = new \DateTime();
+        $folder_made_name = $actual_banner_name . "Folder" . $currentDate->format('Y-m-d_H-i-s');
```

Comment:
I would agree with William's suggestions. You can replace "Folder" with an underscore or something.